AirCasting/Graph/Graph.swift

```diff
@@ -213,7 +238,61 @@ struct Graph: UIViewRepresentable {
             let chartDataEntry = ChartDataEntry(x: timeInterval, y: measurement.value)
             return chartDataEntry
         }) ?? []
-        uiView.updateWithEntries(entries: entries, isAutozoomEnabled: isAutozoomEnabled)
+        
+        if entries.count > 1000 {
+            let simplifiedPoints = SwiftSimplify.simplify(entries,
+                                                          tolerance: 0.000000001,
+                                                          highestQuality: true)
+            uiView.updateWithEntries(entries: simplifiedPoints, isAutozoomEnabled: isAutozoomEnabled)
+            print("Simplified \(entries.count) to \(simplifiedPoints.count)")
+        } else {
+            uiView.updateWithEntries(entries: entries, isAutozoomEnabled: isAutozoomEnabled)
+        }
+         
+        
+        let limitLines = getLimitLines()
+        uiView.updateMidnightLines(with: limitLines)
+    }
+    
+    func getMidnightsPoints(startingDate: Date) -> [Double] {
+        let now = Date()
+        var midnight = Calendar.current.date(bySettingHour: 0, minute: 0, second: 0, of: now)!
+        let components = Calendar.current.dateComponents([.hour, .minute, .second], from: midnight)
+        let firstMeasurementDate = startingDate
+        var midnightPoints: [Double] = []
+        
+        while (midnight > firstMeasurementDate) {
+            let previus = Calendar.current.nextDate(after: midnight,
+                                                    matching: components,
+                                                    matchingPolicy: Calendar.MatchingPolicy.previousTimePreservingSmallerComponents,
+                                                    repeatedTimePolicy: Calendar.RepeatedTimePolicy.first,
+                                                    direction: .backward)
+            
+            let formatter = DateFormatter()
+            formatter.dateStyle = .short
+            formatter.timeStyle = .short
```

Comment:
Still the date formatter is unused and initialised every second

Reply:
Fixed it in https://github.com/HabitatMap/AirCastingiOS/pull/54